packages/block-library/src/footnotes/index.php

```diff
@@ -77,6 +72,24 @@ function register_block_core_footnotes() {
 }
 add_action( 'init', 'register_block_core_footnotes' );
 
+/**
+ * Filters blocks to rewrite footnote links to be popovers instead.
+ *
+ * @param string $block_content Block content.
+ * @return string Filtered block content.
+ */
+function filter_blocks_to_rewrite_footnote_links( $block_content ) {
+	// TODO: Use the HTML API instead!
+	$block_content = preg_replace_callback(
+		':(?P<start_tag><sup data-fn="(?P<fn>[^"]+?)" class="fn">)<a[^>]+?>(?P<num>\d+)</a>(?P<end_tag></sup>):',
+		static function ( $matches ) {
+			return $matches['start_tag'] . sprintf( '<button popovertarget="%1$s" style="anchor-name: --fn-%1$s;">%2$s</button>', esc_attr( $matches['fn'] ), $matches['num'] ) . $matches['end_tag'];
+		},
+		$block_content
+	);
```

Comment:
- [ ] Since `WP_HTML_Tag_Processor` cannot be used to replace nodes, we can subclass `WP_HTML_Tag_Processor` (e.g. anonymously) to get access to the underlying lexical updates.